src/py/covidr/app.py

```diff
@@ -1,25 +1,91 @@
 import os
+from functools import wraps
 from pathlib import Path
+from urllib.parse import urlencode
 
+from authlib.integrations.flask_client import OAuth
 from covidr.config import DevelopmentConfig
-from flask import Flask, send_from_directory
+from covidr.config.config import Auth0Config
+from flask import Flask, redirect, send_from_directory, session, url_for
 
 static_folder = Path("static")
 
 # EB looks for an 'application' callable by default.
 application = Flask(__name__, static_folder=str(static_folder))
 
-
 if os.environ.get("FLASK_ENV") == "development":
     application.config.from_object(DevelopmentConfig())
 
 
+auth0_config = Auth0Config()
+application.secret_key = auth0_config.SECRET_KEY
+
+
+oauth = OAuth(application)
+
+auth0 = oauth.register(
+    "auth0",
+    client_id=auth0_config.AUTH0_CLIENT_ID,
+    client_secret=auth0_config.AUTH0_CLIENT_SECRET,
+    api_base_url=auth0_config.AUTH0_BASE_URL,
+    access_token_url=auth0_config.ACCESS_TOKEN_URL,
+    authorize_url=auth0_config.AUTHORIZE_URL,
+    client_kwargs=auth0_config.CLIENT_KWARGS,
+)
+
+
+# use this to wrap protected views
+def requires_auth(f):
+    @wraps(f)
+    def decorated(*args, **kwargs):
+        if "profile" not in session:
+            # Redirect to Login page
+            return redirect("/login")
+        return f(*args, **kwargs)
+
+    return decorated
```

Comment:
Not necessary to do in this PR, but putting this here could potentially cause problems with recursive imports if we break up `app.py`.